cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/worker/TaskWorkerHttpHandlerInternal.java

```diff
@@ -95,4 +104,38 @@ private String exceptionToJson(Exception ex) {
     BasicThrowable basicThrowable = new BasicThrowable(ex);
     return GSON.toJson(basicThrowable);
   }
+
+  private static class RunnableTaskBodyProducer extends BodyProducer {
```

Comment:
It is good to have a javadoc to explain why this is needed instead of just sending out the `byte[]`.